src/main/java/net/revelc/code/formatter/FormatterMojo.java

```diff
@@ -644,10 +644,16 @@ protected void doFormatFile(File file, ResultCollector rc, Properties hashCache,
         if (!Result.SKIPPED.equals(result)) {
             if (formattedCode == null) {
                 result = Result.FAIL;
-            } else if (originalCode.equals(formattedCode)) {
-                result = Result.SKIPPED;
             } else {
-                result = Result.SUCCESS;
+                // Process the source one more time and remove any trailing whitespace found
+                if (removeTrailingWhitespace) {
+                    formattedCode = REMOVE_TRAILING_PATTERN.matcher(formattedCode).replaceAll("");
+                }
+                if (originalCode.equals(formattedCode)) {
+                    result = Result.SKIPPED;
+                } else {
+                    result = Result.SUCCESS;
+                }
```

Comment:
Replace 5 lines with 1.

```suggestion
                result = originalCode.equals(formattedCode) ? Result.SKIPPED : Result.SUCCESS;
```